hack/lib/golang.sh

```diff
@@ -382,11 +382,14 @@ kube::golang::is_statically_linked_library() {
 kube::golang::binaries_from_targets() {
   local target
   for target; do
-    # If the target starts with what looks like a domain name, assume it has a
-    # fully-qualified package name rather than one that needs the Kubernetes
-    # package prepended.
     if [[ "${target}" =~ ^([[:alnum:]]+".")+[[:alnum:]]+"/" ]]; then
+      # If the target starts with what looks like a domain name, assume it has a
+      # fully-qualified package name rather than one that needs the Kubernetes
+      # package prepended.
       echo "${target}"
+    elif [[ "${target}" =~ ^vendor/ ]]; then
+      # Strip vendor/ prefix, since we're building in gomodule mode.
+      echo "${target#"vendor/"}"
```

Comment:
this is because we have [external callers](https://grep.app/search?q=vendor/github.com/onsi/ginkgo/ginkgo&filter[repo][0]=kubernetes/test-infra) invoking our build scripts passing in `WHAT=vendor/...`

I'll track cleaning that up once all supported release branches have moved to module builds, but until then, we should continue accepting those WHAT targets and fix them up internally to keep producing the right binaries

Reply:
for the most part those should be ginkgo, and the move to v2 will give us a clean point to switch to not using vendor seeing as v2 isn't merged yet. #109111